### ISSUE TITLE
Dart's DOM facade and innerHtml

### ISSUE THREAD

**Issue by pkozlowski-opensource** [angular/angular#789]
There is a wired thing going on with Dart's DOM facade and the `innerHtml` property:
* `DOM(hasProperty, 'innerHtml')` => false
* `DOM(hasProperty, 'innerHTML')` => true

but:

* `reflector.setter('innerHtml')` => exists
* `reflector.setter('innerHTML')` => null

which basically means that we can't use the same name to test for a property name _and_ lookup a setter in Dart.... 

I'm going to special-case this property for now to unblock #619 but if any person more familiar with Dart could give me a hand with this one, it would be awesome.

https://api.dartlang.org/apidocs/channels/be/dartdoc-viewer/dart:html.DocumentFragment#id_innerHtml

**Comment by pkozlowski-opensource** [comment]
This is no longer relevant after the [recent changes](0fb9f3bd6c6efb471da8069a2c6eb2f843ce8050) to the compiler.
